File: tests/fixtures/safe_secret_names.go
package main

import "os"

// Benign names containing secret-keyword substrings, or low-signal
// keywords with non-secret values. None should be flagged by
// go/no-hardcoded-secret after the word-boundary + value-gate fix.

func config() {
	// Substring false positives.
	author := "Pallets"
	authors := "core team"
	authenticated := "yes"
	authorization := "X-Custom"
	tokenizer := "bert-base-uncased"
	secretarial := "filed"

	// Low-signal + secret-named values, all env-sourced (not literals).
	auth := os.Getenv("AUTH")
	token := os.Getenv("TOKEN")
	password := os.Getenv("PW")
	apiKey := os.Getenv("API_KEY")
	secretKey := os.Getenv("SECRET_KEY")

	_ = author
	_ = authors
	_ = authenticated
	_ = authorization
	_ = tokenizer
	_ = secretarial
	_ = auth
	_ = token
	_ = password
	_ = apiKey
	_ = secretKey
}

File: tests/fixtures/safe_secret_names.cs
// Benign names containing secret-keyword substrings, or low-signal
// keywords with env-sourced values. None should be flagged by
// cs/no-hardcoded-secret after the word-boundary + value-gate fix.
using System;

public class SafeSecretNames
{
    // Substring false positives.
    const string author = "Pallets";
    const string authors = "core team";
    const string authenticated = "yes";
    const string authorizationScheme = "Bearer";
    const string tokenizer = "bert-base-uncased";
    const string secretarialNote = "filed";

    static void Load()
    {
        // Low-signal + secret-named values, all env-sourced (not literals).
        string auth = Environment.GetEnvironmentVariable("AUTH");
        string token = Environment.GetEnvironmentVariable("TOKEN");
        string password = Environment.GetEnvironmentVariable("PW");
        string apiKey = Environment.GetEnvironmentVariable("API_KEY");
        string connectionString = Environment.GetEnvironmentVariable("CONN");
        Console.WriteLine(auth + token + password + apiKey + connectionString);
    }
}
